### PR TITLE
Add the section: Constraint Satisfaction Problem (CSP) Solver

### PR DESCRIPTION
**Pull request by alessandroleite**
* Add some well-known off-the-shelf constraint satisfaction problem solvers.
* Include the projects: JScience, JADE and Protégé to Utility section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,6 +247,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [Eclipse](http://www.eclipse.org/) - Does a lot of things in the background. Noteworthy are its large amount of plugins.
 * [IntelliJ IDEA](http://www.jetbrains.com/idea/) - Supports a lot of JVM languages and provides good options for Android development. The commercial edition targets the enterprise sector.
+* [JDeveloper](http://www.oracle.com/technetwork/developer-tools/jdev/overview/index-094652.html) - Providers a complete environment to write Java SE and EE applications, as well as to develop SOA applications based on the [Oracle SOA Suite](http://www.oracle.com/us/products/middleware/soa/suite/overview/index.html).
```
Comment (akullpp) [inline README.md]: Absolutely not. JDeveloper is the epitome of what this list isn't about.

File: README.md [hunk 2]
```diff
@@ -480,6 +492,7 @@ h
 * [Google Web Toolkit](http://www.gwtproject.org/) - Toolbox which includes a Java-to-JavaScript compiler for client-side code, XML parser, API for RPC, JUnit integration, internationalization support and widgets for the GUI.
 * [Grails](https://grails.org/) - Groovy framework with the aim to provide a highly productive environment by favoring convention over configuration, no XML and support for mixins.
 * [Ninja](http://www.ninjaframework.org/) - Full stack web framework for Java. Rock solid, fast and super productive.
+* [Oracle ADF](http://www.oracle.com/technetwork/developer-tools/adf/overview/index.html) - Providers a free and a commercial version framework to write web and mobile applications, following an event-driven approach.
```
Comment (akullpp) [inline README.md]: This list will never include the ADF framework. It is the exact opposite of the definition awesome lists have.

Comment (alessandroleite) [inline README.md]: Ok. I'm curious why Netbeans and PrimeFaces are awesome and JDeveloper and Oracle ADF are disgusting.

Comment (kkrauth) [inline README.md]: JDeveloper comes prepackaged with chock full of modules and plugins for working with Oracle specific technologies (i.e. ADF). It's really an enterprise-style development environment that I would also argue is not appropriate for this list. 

Equally, I have never heard anyone using ADF for a personal project, nor would I recommend anyone to use it on any kind of non-enterprise project, so to compare it with PrimeFaces is a bit disingenuous.

File: README.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
     - [Code Analysis](#code-analysis)
     - [Compiler-compiler](#compiler-compiler)
     - [Configuration](#configuration)
+    - [Constraint Satisfaction Problem (CSP) Solver](#constraint-satisfaction-problem-csp-solver)
```
Comment (akullpp) [inline README.md]: Please remove the abbreviation from the title.

File: README.md [hunk 4]
```diff
@@ -100,6 +101,13 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [config](https://github.com/typesafehub/config) - Configuration library for JVM languages.
 
+## Constraint Satisfaction Problem (CSP) Solver
```
Comment (akullpp) [inline README.md]: Please remove the abbreviation from the title.

Comment (alessandroleite) [inline README.md]: It's done.

File: README.md [hunk 5]
```diff
@@ -100,6 +101,13 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [config](https://github.com/typesafehub/config) - Configuration library for JVM languages.
 
+## Constraint Satisfaction Problem Solver
```
Comment (akullpp) [inline README.md]: Can you add a short description just as the other sections have?

**Comment by akullpp** [comment]
Thanks a lot.

**Pull request merged**